rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake

```diff
@@ -27,53 +27,60 @@ rosidl_generate_dds_interfaces(
 
 set(_dds_idl_files "")
 set(_dds_idl_base_path "${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_dds_idl")
-set(_dds_idl_path "${_dds_idl_base_path}/${PROJECT_NAME}/dds_opensplice")
 foreach(_idl_file ${rosidl_generate_interfaces_IDL_FILES})
   get_filename_component(_extension "${_idl_file}" EXT)
-  if("${_extension}" STREQUAL ".msg")
-    get_filename_component(name "${_idl_file}" NAME_WE)
-    list(APPEND _dds_idl_files "${_dds_idl_path}/${name}_.idl")
-  elseif("${_extension}" STREQUAL ".srv")
-    get_filename_component(name "${_idl_file}" NAME_WE)
-    list(APPEND _dds_idl_files "${_dds_idl_path}/Sample${name}Request_.idl")
-    list(APPEND _dds_idl_files "${_dds_idl_path}/Sample${name}Response_.idl")
+  get_filename_component(_name "${_idl_file}" NAME_WE)
+  if("${_extension} " STREQUAL ".msg ")
+    get_filename_component(_parent_folder "${_idl_file}" DIRECTORY)
+    get_filename_component(_parent_folder "${_parent_folder}" NAME)
+    list(APPEND _dds_idl_files
+      "${_dds_idl_base_path}/${PROJECT_NAME}/${_parent_folder}/dds_opensplice/${_name}_.idl")
```

Comment:
That is exactly what is happening here: `_parent_folder` contains either `msg` or `srv` depending on if it is a plain message or a request / response message.